docs/rechunking.ipynb

```diff
@@ -0,0 +1,418 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "WARNING:apache_beam.options.pipeline_options:Discarding unparseable args: ['/Users/shoyer/miniconda3/envs/xarray-beam/lib/python3.9/site-packages/ipykernel_launcher.py', '-f', '/Users/shoyer/Library/Jupyter/runtime/kernel-2b08a4f7-b064-490e-9ac1-e22ceca4c6cd.json']\n",
+      "WARNING:root:Make sure that locally built Python SDK docker image has Python 3.9 interpreter.\n",
+      "WARNING:apache_beam.coders.coder_impl:Using fallback deterministic coder for type '<class 'xarray_beam._src.core.Key'>' in 'Rechunk/Stage0/Consolidate/GroupByTempKeys'. \n",
+      "WARNING:apache_beam.coders.coder_impl:Using fallback deterministic coder for type '<class 'xarray_beam._src.core.Key'>' in 'Rechunk/Stage0/Consolidate/GroupByTempKeys'. \n",
+      "WARNING:apache_beam.coders.coder_impl:Using fallback deterministic coder for type '<class 'xarray_beam._src.core.Key'>' in 'Rechunk/Stage2/Consolidate/GroupByTempKeys'. \n",
+      "WARNING:apache_beam.coders.coder_impl:Using fallback deterministic coder for type '<class 'xarray_beam._src.core.Key'>' in 'Rechunk/Stage2/Consolidate/GroupByTempKeys'. \n"
+     ]
+    },
+    {
+     "data": {
+      "text/plain": [
+       "[(Key(offsets={'x': 0, 'y': 0}, vars=None),\n",
+       "  <xarray.Dataset>\n",
+       "  Dimensions:  (x: 5, y: 2)\n",
+       "  Dimensions without coordinates: x, y\n",
+       "  Data variables:\n",
+       "      foo      (x, y) int64 0 1 2 3 4 5 6 7 8 9\n",
+       "      bar      (x, y) int64 100 101 102 103 104 105 106 107 108 109),\n",
+       " (Key(offsets={'x': 5, 'y': 0}, vars=None),\n",
+       "  <xarray.Dataset>\n",
+       "  Dimensions:  (x: 3, y: 2)\n",
+       "  Dimensions without coordinates: x, y\n",
+       "  Data variables:\n",
+       "      foo      (x, y) int64 10 11 12 13 14 15\n",
+       "      bar      (x, y) int64 110 111 112 113 114 115)]"
+      ]
+     },
+     "execution_count": 7,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "inputs | xbeam.Rechunk(dim_sizes={'x': 6}, source_chunks={'x': 3}, target_chunks={'x': 5}, itemsize=8)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "`Rechunk` requires specifying a few more parameters, but based on that information it can be _much_ more efficient for more complex rechunking tasks, particular in cases where data needs to be distributed into a very different shape (e.g., distributing a matrix across rows vs. columns). A naive \"splitting\" approach in such cases could divide datasets into extremely small tasks corresponding to individual array elements, which adds a huge amount of overhead."
```

Comment:
I really like the advice that you've mentioned to try to make chunk sizes operate in the 10s-100s of MB range. I think readings of the tutorial would like that kind of nudge. 

Better still, this kind an explanation while referencing [the XArray optimization tips](http://xarray.pydata.org/en/stable/user-guide/dask.html#optimization-tips) would be grand.

Reply:
Reading more of the examples, it would be cool to see how to think about splitting vars for optimization purposes.

Reply:
I added a new section on "Choosing chunks".

Note that most of the Xarray optimization tips (beyond chunk sizes) don't directly apply to using Xarray-Beam. This is worth highlighting somewhere, e.g., when I move the discussion of pros/cons vs Dask from the README into the dosc.